service/controller/app/resource/chartoperator/create_test.go

```diff
@@ -0,0 +1,414 @@
+package chartoperator
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/app/v6/pkg/key"
+	"github.com/giantswarm/app/v6/pkg/values"
+	"github.com/giantswarm/helmclient/v4/pkg/helmclient"
+	"github.com/giantswarm/helmclient/v4/pkg/helmclienttest"
+	"github.com/giantswarm/k8sclient/v6/pkg/k8sclienttest"
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/giantswarm/app-operator/v5/service/controller/app/controllercontext"
+)
+
+func Test_Resource_EnsureCreated(t *testing.T) {
+	tests := []struct {
+		name             string
+		apps             []runtime.Object
```

Comment:
Using `runtime.Object` is a bit strange here. Could we use the actual types and then convert them when setting up the fake client?